akka-docs/rst/java/code/docs/http/javadsl/server/directives/BasicDirectivesExamplesTest.java

```diff
@@ -0,0 +1,788 @@
+  }
+
+  @Test
+  public void testExtractExecutionContext() {
+    //#extractExecutionContext
+    final Route route = path("sample", () ->
+      extractExecutionContext(executor ->
+        onSuccess(() ->
+          CompletableFuture.supplyAsync(() ->
+            "Run on " + executor.hashCode() + "!", executor // uses the `executor` ExecutionContext
+          ), str -> complete(str)
+        )
+      )
+    );
+
+    //tests:
+    testRoute(route).run(HttpRequest.GET("/sample"))
+      .assertEntity("Run on " + system().dispatcher().hashCode() + "!");
+    //#extractExecutionContext
+  }
+
+  @Test
+  public void testWithLog() {
+    //#withLog
+    final LoggingAdapter special = Logging.getLogger(system(), "SpecialRoutes");
+
+    final Route sample = path("sample", () ->
+      extractLog(log -> {
+        final String msg = "Logging using " + log + "!";
+        log.debug(msg);
+        return complete(msg);
+      }
+      )
+    );
+
+    final Route route = route(
+      pathPrefix("special", () ->
+        withLog(special, () -> sample)
+      ),
+      sample
+    );
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/sample"))
+      .assertEntity("Logging using " + system().log() + "!");
+    testRoute(route).run(HttpRequest.GET("/special/sample"))
+      .assertEntity("Logging using " + special + "!");
+    //#withLog
+  }
+
+  @Ignore("Ignore compile-only test")
+  @Test
+  public void testWithSettings() {
+    //#withSettings
+    final RoutingSettings special =(RoutingSettings) RoutingSettings.create(system().settings().config()).withFileIODispatcher("special-io-dispatcher");
+
+    final Route sample = path("sample", () -> {
+      // internally uses the configured fileIODispatcher: ContentTypes.APPLICATION_JSON, source
+      final Source<ByteString, Object> source =
+        FileIO.fromPath(Paths.get("example.json")).mapMaterializedValue(completionStage -> (Object) completionStage);
+      return complete(
+        HttpResponse.create().withEntity(HttpEntities.create(ContentTypes.APPLICATION_JSON, source))
+      );
+    });
+
+    final Route route = get(() ->
+      route(
+        pathPrefix("special", () ->
+          withSettings(special, () -> sample) // `special` file-io-dispatcher will be used to read the file
+        ),
+        sample // default file-io-dispatcher will be used to read the file
+      )
+    );
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/special/sample"))
+      .assertEntity("{}");
+    testRoute(route).run(HttpRequest.GET("/sample"))
+      .assertEntity("{}");
+    //#withSettings
+  }
+
+  @Test
+  public void testMapResponse() {
+    //#mapResponse
+    final Route route = mapResponse(
+      response -> response.withStatus(StatusCodes.BAD_GATEWAY),
+      () -> complete("abc")
+    );
+
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/abcdef?ghi=12"))
+      .assertStatusCode(StatusCodes.BAD_GATEWAY);
+    //#mapResponse
+  }
+
+  class ApiRoute {
```

Comment:
Make the fields private final, do not assign `null`, change "_system" to "system" (reference using `this.system = system` in constructor.

Should be marked up to be included in the directive docs like the Scala one is